src/sagemaker/jumpstart/curated_hub/content_copy.py

```diff
@@ -0,0 +1,289 @@
+
+        return copy_configs
+
+    def _get_copy_configs_for_training_dataset(self, model_specs: JumpStartModelSpecs) -> None:
+        src_prefix = self._src_s3_filesystem.get_default_training_dataset_s3_reference(model_specs)
+        dst_prefix = self._dst_s3_filesystem.get_default_training_dataset_s3_reference(model_specs)
+
+        keys_in_src_dir = self._get_s3_object_keys_under_prefix(src_prefix)
+
+        copy_configs = []
+        for full_key in keys_in_src_dir:
+            src_reference = create_s3_object_reference_from_bucket_and_key(
+                src_prefix.bucket, full_key
+            )
+            dst_reference = create_s3_object_reference_from_bucket_and_key(
+                dst_prefix.bucket, full_key.replace(src_prefix.key, dst_prefix.key, 1)
+            )  # Replacing s3 key prefix with expected destination prefix
+
+            copy_configs.append(
+                CopyContentConfig(
+                    src=src_reference, dst=dst_reference, display_name="training dataset"
+                )
+            )
+
+        return copy_configs
+
+    def _get_s3_object_keys_under_prefix(self, prefix_reference: S3ObjectReference) -> Set[str]:
+        try:
+            return find_objects_under_prefix(
+                bucket=prefix_reference.bucket,
+                prefix=prefix_reference.key,
+                s3_client=self._s3_client,
+            )
+        except Exception as ex:
+            print(
+                "ERROR: encountered an exception when finding objects"
+                + f" under prefix {prefix_reference.bucket}/{prefix_reference.key}: {str(ex)}"
+            )
+            raise ex
+
+    def _get_copy_configs_for_demo_notebook_dependencies(
+        self, model_specs: JumpStartModelSpecs
+    ) -> None:
+        copy_configs = []
+
+        notebook_s3_reference = self._src_s3_filesystem.get_demo_notebook_s3_reference(model_specs)
+        notebook_s3_reference_dst = self._dst_s3_filesystem.get_demo_notebook_s3_reference(
+            model_specs
+        )
+        copy_configs.append(
+            CopyContentConfig(
+                src=notebook_s3_reference,
+                dst=notebook_s3_reference_dst,
+                display_name="demo notebook",
+            )
+        )
+
+        return copy_configs
+
+    def _get_copy_configs_for_markdown_dependencies(self, model_specs: JumpStartModelSpecs) -> None:
+        copy_configs = []
+
+        markdown_s3_reference = self._src_s3_filesystem.get_markdown_s3_reference(model_specs)
+        markdown_s3_reference_dst = self._dst_s3_filesystem.get_markdown_s3_reference(model_specs)
+        copy_configs.append(
+            CopyContentConfig(
+                src=markdown_s3_reference, dst=markdown_s3_reference_dst, display_name="markdown"
+            )
+        )
+
+        return copy_configs
+
+    def _parallel_execute_copy_configs(self, copy_configs: List[CopyContentConfig]):
```

Comment:
question: perhaps we want to put this copying util in a higher level utility, so other modules may use it?